mods/d2k/maps/harkonnen-07/harkonnen07.lua

```diff
@@ -0,0 +1,238 @@
+--[[
+   Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+
+AtreidesMainBase = { AConYard1, AOutpost1, ARefinery1, ARefinery2, AHeavyFactory1, ALightFactory1, ALightFactory2, AHiTechFactory, AResearch, ARepair, AStarport, AGunt1, AGunt2, AGunt3, AGunt4, AGunt5, AGunt6, ARock1, ARock2, ARock3, ARock4, ARock5, ARock6, APower1, APower2, APower3, APower4, APower5, APower6, APower7, APower8, APower9, APower10, APower11, APower12, APower13, APower14, APower15, APower16, APower17, APower18, APower19, APower20, APower21, APower22, ASilo1, ASilo2, ASilo3 }
+AtreidesSmallBase = { AOutpost2, ARefinery3, ABarracks, AHeavyFactory2, AGunt7, AGunt8, APower23, APower24, APower25, APower26, ASilo4, ASilo5, ASilo6, ASilo7 }
+CorrinoBase = { CBarracks, COutpost, CPower1, CPower2, CPower3, CRock1, CRock2 }
+
+AtreidesReinforcements =
+{
+	easy = { "sonic_tank" },
+	normal = { "sonic_tank", "trooper" },
+	hard = { "sonic_tank", "quad" }
+}
+
+CorrinoReinforcements =
+{
+	easy =
+	{
+		{ "sardaukar", "sardaukar", "sardaukar", "quad", "trike", "combat_tank_h" },
+		{ "sardaukar", "sardaukar", "combat_tank_h", "siege_tank" }
+	},
+
+	normal =
+	{
+		{ "sardaukar", "sardaukar", "sardaukar", "quad", "trike", "trike", "combat_tank_h" },
+		{ "sardaukar", "sardaukar", "sardaukar", "combat_tank_h", "siege_tank" }
+	},
+
+	hard =
+	{
+		{ "sardaukar", "sardaukar", "sardaukar", "quad", "quad", "trike", "trike", "combat_tank_h" },
+		{ "sardaukar", "sardaukar", "sardaukar", "sardaukar", "combat_tank_h", "siege_tank" }
+	}
+}
+
+EnemyAttackDelay =
+{
+	easy = DateTime.Minutes(2) + DateTime.Seconds(30),
+	normal = DateTime.Minutes(2),
+	hard = DateTime.Minutes(1) + DateTime.Seconds(30)
+}
+
+InitialAtreidesReinforcements =
```

Comment:
Could you make this difficulty-dependant too, and tone it down slightly for normal and considerably for easy?